src/lib/auth.ts

```diff
@@ -40,39 +63,141 @@ export const ROLES: {
 	],
 	// also default-user (see below)
 	'named-user-api-key': [
-		'resin.actor.delete?id eq @__ACTOR_ID',
-		'resin.api_key.read?is_of__actor eq @__ACTOR_ID',
-		'resin.application.all',
-		'resin.device_type.read',
-		'resin.device_type_alias.read',
-		'resin.cpu_architecture.read',
-		'resin.application_config_variable.all',
-		'resin.application_environment_variable.all',
-		'resin.application_tag.all',
-		'resin.application_type.all',
-		'resin.device.all',
-		'resin.device.tunnel-22222',
-		'resin.device_config_variable.all',
-		'resin.device_environment_variable.all',
-		'resin.device_tag.all',
-		'resin.device_service_environment_variable.all',
-		'resin.image.all',
-		'resin.image__is_part_of__release.all',
-		'resin.image_environment_variable.all',
-		'resin.image_install.all',
-		'resin.image_label.all',
-		'resin.organization.read',
-		'resin.organization_membership.read',
-		'resin.release.all',
-		'resin.release_tag.all',
-		'resin.service.all',
-		'resin.service_environment_variable.all',
-		'resin.service_install.all',
-		"resin.service_instance.read?service_type eq 'vpn'",
-		'resin.service_label.all',
-		'resin.user.read',
-		`resin.user__has__public_key.all?${matchesUser}`,
-		'resin.release_asset.all',
+		// 'resin.actor.delete?id eq @__ACTOR_ID',
+		resinAuth({
+			resource: 'actor',
+			access: 'delete',
+			options: {
+				$filter: {
+					id: {
+						'@': '__ACTOR_ID',
+					},
+				},
+			},
+		}),
+
+		// 'resin.api_key.read?is_of__actor eq @__ACTOR_ID',
+		resinAuth({
+			resource: 'api_key',
+			access: 'read',
+			options: {
+				$filter: {
+					is_of__actor: {
+						'@': '__ACTOR_ID',
+					},
+				},
+			},
+		}),
+
+		// 'resin.application.all',
+		resinAuth({ resource: 'application', access: 'all' }),
```

Comment:
Things are fairly verbose like this, I think it's better though and worth adding the verbosity but I do wonder if in future we'll add shortcuts to reduce the verbosity slightly whilst keeping the benefits. For now I think I'd say live with the minor pain for the greater good it brings

Reply:
I this this can be easily reduced by making the parameters on the return warpper 3 values e.g.
`balenaAuth('application', 'all')`